src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java

```diff
@@ -0,0 +1,149 @@
+        assertFalse(result.isPresent());
+    }
+
+    @Test
+    void findAllWhenMapIsEmpty() {
+        List<Payment> paymentList = inMemPaymentRepository.findAll();
+
+        assertEquals(0, paymentList.size());
+    }
+
+    @Test
+    void findAllWhenMapContainsOnePayment() {
+        Payment payment = createPayment();
+        inMemPaymentRepository.save(payment);
+
+        List<Payment> paymentList = inMemPaymentRepository.findAll();
+
+        assertAll(
+                () -> assertEquals(1, paymentList.size()),
+                () -> assertTrue(paymentList.contains(payment))
+        );
+    }
+
+    @Test
+    void saveWhenPaymentIsNull() {
+        assertThrows(IllegalArgumentException.class,
+                () -> inMemPaymentRepository.save(null),
+                "Payment must not be null");
+    }
+
+    @Test
+    void saveWhenPaymentIsNotContained() {
+        Payment payment = mock(Payment.class);
+        Payment result = inMemPaymentRepository.save(payment);
+
+        assertEquals(payment, result);
+    }
+
+    @Test
+    void saveWhenPaymentIsContained() {
+        Payment payment = createPayment();
+        Payment result = inMemPaymentRepository.save(payment);
+
+        assertAll(
+                () -> assertEquals(payment, result),
+                () -> assertThrows(IllegalArgumentException.class,
+                        () -> inMemPaymentRepository.save(payment),
+                        "Payment with id " + payment.getPaymentId() + "already saved")
+        );
+    }
+
+    @Test
+    void editMessageWhenPaymentIdIsNull() {
+        assertThrows(NoSuchElementException.class,
+                () -> inMemPaymentRepository.editMessage(null, "Some message"));
+    }
+
+    @Test
+    void editMessageWhenPaymentIdIsNotContained() {
+        savingThreeMocks();
+        UUID paymentId = UUID.randomUUID();
+
+        assertThrows(NoSuchElementException.class,
+                () -> inMemPaymentRepository.editMessage(paymentId, "Some message"),
+                "Payment with id " + paymentId + " not found");
+    }
+
+    @Test
+    void editMessageWhenPaymentIdIsContained() {
+        Payment payment = createPayment();
+        savingTwoMocksAndOnePayment(payment);
+
+        Payment result = inMemPaymentRepository.editMessage(payment.getPaymentId(), "New message");
+
+        assertAll(
+                () -> assertEquals(payment, result),
+                () -> assertEquals("New message", result.getMessage())
+        );
+    }
+
+    private Payment createPayment() {
+        return new Payment(1, 1D, "Old message");
+    }
+
+    private void savingTwoMocksAndOnePayment(Payment payment) {
+        inMemPaymentRepository.save(mock(Payment.class));
+        inMemPaymentRepository.save(mock(Payment.class));
+        inMemPaymentRepository.save(payment);
+    }
+
+    private void savingThreeMocks() {
+        inMemPaymentRepository.save(mock(Payment.class));
+        inMemPaymentRepository.save(mock(Payment.class));
```

Comment:
create a separate method that takes an argument as number of payments you want to save and call it here and in previous method like savePaymentMocks(3) or savePaymentMocks(2);

Reply:
Fixed.